runtime/pendulum/src/xcm_config.rs

```diff
@@ -92,6 +100,45 @@ impl Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert {
 	}
 }
 
+pub struct RelativeValue {
+	num: Balance,
+	denominator: Balance,
+}
+
+impl RelativeValue {
+	fn adjust_amount_by_relative_value(amount: Balance, relative_value: RelativeValue) -> Balance {
+		if relative_value.denominator == 0 {
+			// Or probably error
+			return amount
+		}
+		// Calculate the adjusted amount
+		let adjusted_amount = amount * relative_value.denominator / relative_value.num;
```

Comment:
Shouldn't we rather be defensive here and use functions like [saturating_mul](https://docs.rs/sp-runtime/latest/sp_runtime/traits/trait.Saturating.html#) and [checked_div](https://docs.rs/sp-runtime/latest/sp_runtime/traits/trait.CheckedDiv.html) to prevent the runtime from panicking?

Also, I think it's a bit counter-intuitive to multiply with a denominator and divide by a numerator although it works. Don't have a suggestion for better naming though.

Reply:
Yes I overlook the saturating operators, I think it is a good idea to add it. I can append the changes to [this PR](https://github.com/pendulum-chain/pendulum/pull/386).

And about the naming, it can be confusing on the operation, but on the definition of the rational number for the value ratio makes sense to call them that way.

Reply:
I added the changes [here](https://github.com/pendulum-chain/pendulum/pull/386/files#diff-3287a4ee060b1070a8e6559a094b4890a596a1b281effef54cafe59da20edf49R79). Also moved the type `RelativeValue` itself to common.